### PR TITLE
airframe-http: #1131 Add a sync client using URLConnection 

### PR DESCRIPTION
**Pull request by xerial**
This PR is for adding a handy http client implementation based on java.net.URLConnection, which doesn't require extra dependencies.

- [x] Add Http client implementation using URLConnection
- [x] Add URLConnection specific exception handling
- [x] Add test cases
- [x] Add factory methods
- [x] Add http client documentations

If we can use jdk11, https://openjdk.java.net/groups/net/httpclient/intro.html seems better.

### CODE REVIEW

**Comment by xerial** [comment]
@takezoe PTAL.  Although URLConnection is handy to use, a catch was it doesn't support PATCH (which is used in some of our Operate API method). 

If supporting only JDK11 is acceptable, we should support JDK11's HTTP client backend (HttpClientBackend is an abstraction for adding such implementation later)

**Comment by takezoe** [comment]
Spark related things will need Java8 support for a while. However, using JDK11's HTTP client makes sense. We can build full-featured Airframe HTTP client implementation on it.

File: airframe-control/src/main/scala/wvlet/airframe/control/IO.scala [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  def readAsString(url: URL): String = {
+    withResource(url.openStream()) { in => readAsString(in) }
+  }
+
+  def readAsString(in: InputStream): String = {
+    readFully(in) { data => new String(data, StandardCharsets.UTF_8) }
+  }
+
+  def readFully[U](in: InputStream)(f: Array[Byte] => U): U = {
```
Comment (takezoe) [inline airframe-control/src/main/scala/wvlet/airframe/control/IO.scala:38]: Does this method need to take a function? How about simply returning `Array[Byte]`?

Comment (xerial) [reply airframe-control/src/main/scala/wvlet/airframe/control/IO.scala:38]: That's a good idea. I'll fix it.

**Pull request merged**